src/test/java/org/broadinstitute/hellbender/utils/read/AlignmentUtilsUnitTest.java

```diff
@@ -141,21 +141,59 @@ private GATKRead makeReadForAlignedToRefTest(final String baseString) {
             badHap.setAlignmentStartHapwrtRef(hapStart);
 
             final int expectedPos = 10130740;
-            tests.add(new Object[]{read, badHap, refStart, expectedPos, goodCigar});
+            tests.add(new Object[]{read, badHap, badHap, refStart, expectedPos, goodCigar});
         }
 
+        // example where left-align generates a leading deletion
+        {
+            final String ref = "CTGAACGTAACCAAAATCAATATGGATACTGAGAAATACTATTTAATAAAGACATAAATTAGACTGCTAAAAAAAATTAAAGAAATTTCAAAAGAGAATCCACCTCTTTTCCTTGCCAGTGCTCAAAAGTGAGTGTGAATCTGGTGGCTGTGGGGCTGTTTTTGGTGTGGCTCTTTGGACCAGCCTGCCTGGTAATTCAAGCCTGCCTCTCATTTCTG";
+            final String hap = "CTGAACGTAACCAAAATCAATATGGATACTGAGAAATACTATTTAATAAAGACATAAATTAGACTGCTAAAAAAAATTAAAGAAATTTCAAAAGAGAATCCACCTCTTTTCCTTGCCAGTGCTCAAAAGTGAGTGTGAATCTGGTGGCTGCGGGGCTGTTTTTGGTGTGGCTCTTTGGACCAGCCTGCCTGGTAATTCAAGCCTGCCTCTCATTTCTG";
+            final String hapCigar = "218M";
+            final String readBases = "GCTGCTTTTGGTGTGGCTCTTT";
+            final GATKRead read = makeReadForAlignedToRefTest(readBases);
+            final int refStart = 215239171;
+            final int hapStart = 575;
+            final String goodCigar = "22M";
+            final Haplotype badHap = new Haplotype(hap.getBytes());
+            badHap.setCigar(TextCigarCodec.decode(hapCigar));
+            badHap.setAlignmentStartHapwrtRef(hapStart);
+            final Haplotype refHap = makeHaplotypeForAlignedToRefTest(ref, ref.length() + "M");
+
+            final int expectedPos = 215239900;
```

Comment:
Could you please add a comment explaining why this is the expected read start position in this case?

Reply:
Done.